src/main/java/com/idrsolutions/microservice/BaseServletContextListener.java

```diff
@@ -241,4 +242,34 @@ private void validateFileDeletionServiceFrequency(final Properties properties) {
             LOG.log(Level.WARNING, message);
         }
     }
+
+    private static void validateMaxConversionDuration(final Properties properties) {
+        final String maxDuration = properties.getProperty(KEY_PROPERTY_MAX_CONVERSION_DURATION);
+        if (maxDuration == null || maxDuration.isEmpty() || "0".equals(maxDuration)
+                || (!maxDuration.matches("\\d+") && !"Infinity".equals(maxDuration))) {
+            properties.setProperty(KEY_PROPERTY_MAX_CONVERSION_DURATION, String.valueOf(Long.MAX_VALUE));
+            final String message = String.format("Properties value for \"maxConversionDuration\" was set to " +
+                    "\"%s\" but should be a positive long or Infinity. Using a value of Infinity.", maxDuration);
+            LOG.log(Level.WARNING, message);
+        } else if ("Infinity".equals(maxDuration)) {
+            properties.setProperty(KEY_PROPERTY_MAX_CONVERSION_DURATION, String.valueOf(Long.MAX_VALUE));
+            final String message = String.format("Interpreting \"Infinity\" as %d", Long.MAX_VALUE);
+            LOG.log(Level.INFO, message);
+        }
+    }
+
+
+    private void initialiseDatabase(final String jndi) {
+        if (jndi == null || jndi.isEmpty()) {
+            final String message = "It is recommended to set your own database instead of using the default internal database as it will allow you to more easily scale the service in the future.\n" +
+                    "More details on the benefits and how to do this can be found here https://support.idrsolutions.com/buildvu/tutorials/cloud/options/external-state-database";
```

Comment:
@KieranFrance This is a product specific link to documentation in the base-microservice. Should it be extracted into the product specific microservices?

Reply:
I have moved the change out to the product specific implementations.
I have introduced a flag to make it simple to determine if the internal database is in use